lib/index.ts

```diff
@@ -4,11 +4,42 @@ import * as path from 'path';
 import * as subProcess from './sub-process';
 import { legacyPlugin } from '@snyk/cli-interface';
 import { containsJar, createPomForJar, createPomForJars, isJar } from './jar';
+import * as os from 'os';
 
 export interface MavenOptions extends legacyPlugin.BaseInspectOptions {
   scanAllUnmanaged?: boolean;
 }
 
+export function getCommand(root: string, targetFile: string | undefined) {
+  if (!targetFile) {
+    return 'mvn';
+  }
+  const isWinLocal = /^win/.test(os.platform()); // local check, can be stubbed in tests
+  const wrapperScript = isWinLocal ? 'mvnw.cmd' : './mvnw';
+  // try to find a sibling wrapper script first
+  let pathToWrapper = path.resolve(
+    root,
+    path.dirname(targetFile),
+    wrapperScript,
+  );
+  if (fs.existsSync(pathToWrapper)) {
+    return wrapperScript;
+  }
+  // now try to find a wrapper in the root
+  pathToWrapper = path.resolve(root, wrapperScript);
+  if (fs.existsSync(pathToWrapper)) {
+    return wrapperScript;
+  }
+  return 'mvn';
+}
+
+// When we have `mvn`, we can run the subProcess from anywhere.
+// However due to https://github.com/takari/maven-wrapper/issues/133, `mvnw` can only be run
+// within the directory where `mvnw` exists
+function calculateTargetFilePath(mavenCommand, root: string, targetPath) {
```

Comment:
@orsagie, @lili2311 
fyi, this is will be a limitation because of that ticket I mentioned.
Do you think I should also add a line in the `README.md`?

Reply:
isn't this what cwd is for?

Reply:
yes, it is, but the problem occurs when you specify a relative path to the targetFile will encounter a error as the `mvnw` is found, but the pom.xml is not present in that directory.
So, until that is fixed, you will not be able to specify a relative path for the targeFile (`../somewhere-else`) when `mvnw` is present.

That's what I meant by limitation.

Because at first I just went ahead and `cwd` with where the targetFile path is, but then I bumped into issues where the path to the pom.xml would be messed up while `subProcess` would be ran in both `mvn` and `mvnw` cases.

Reply:
In that case let's add a check to see that `mvnw` is present in the `root` and propagate an error with instructions on how to properly run if that is not the case.

Reply:
will that work if you go with the `targetFile` approach being outside of that directory where you want to test your project?